--- a/js/ui/unlockDialog.js	2022-11-23 22:38:36.353692379 -0500
+++ b/js/ui/unlockDialog.js	2022-11-23 22:38:11.580553201 -0500
@@ -24,9 +24,6 @@ const CROSSFADE_TIME = 300;
 const FADE_OUT_TRANSLATION = 200;
 const FADE_OUT_SCALE = 0.3;
 
-const BLUR_BRIGHTNESS = 0.55;
-const BLUR_SIGMA = 60;
-
 const SUMMARY_ICON_SIZE = 32;
 
 var NotificationsBox = GObject.registerClass({
@@ -524,10 +521,6 @@ var UnlockDialog = GObject.registerClass
 
         this._bgManagers = [];
 
-        const themeContext = St.ThemeContext.get_for_stage(global.stage);
-        themeContext.connectObject('notify::scale-factor',
-            () => this._updateBackgroundEffects(), this);
-
         this._updateBackgrounds();
         Main.layoutManager.connectObject('monitors-changed',
             this._updateBackgrounds.bind(this), this);
@@ -640,7 +633,6 @@ var UnlockDialog = GObject.registerClass
             y: monitor.y,
             width: monitor.width,
             height: monitor.height,
-            effect: new Shell.BlurEffect({ name: 'blur' }),
         });
 
         let bgManager = new Background.BackgroundManager({
@@ -654,21 +646,6 @@ var UnlockDialog = GObject.registerClass
         this._backgroundGroup.add_child(widget);
     }
 
-    _updateBackgroundEffects() {
-        const themeContext = St.ThemeContext.get_for_stage(global.stage);
-
-        for (const widget of this._backgroundGroup) {
-            const effect = widget.get_effect('blur');
-
-            if (effect) {
-                effect.set({
-                    brightness: BLUR_BRIGHTNESS,
-                    sigma: BLUR_SIGMA * themeContext.scale_factor,
-                });
-            }
-        }
-    }
-
     _updateBackgrounds() {
         for (let i = 0; i < this._bgManagers.length; i++)
             this._bgManagers[i].destroy();
@@ -678,7 +655,6 @@ var UnlockDialog = GObject.registerClass
 
         for (let i = 0; i < Main.layoutManager.monitors.length; i++)
             this._createBackground(i);
-        this._updateBackgroundEffects();
     }
 
     _ensureAuthPrompt() {

